Sidebar with Glyphs

ErikKarlen opened the pull request:
Added a new activity bar with all the glyphs found in the snippets file that can be clicked on to add to the active editor. Also made gen_snippets.js pretty print and updated the snippets.code-snippets file. This added snippets for glyphs that don't use backslash to write like + for add.

Picture of sidebar:
<img width="393" height="830" alt="image" src="https://github.com/user-attachments/assets/ef583277-7366-4e5f-9508-bf4bedaa1cfe" />

ErikKarlen committed:
Update snippets.code-snippets

Adds snippets for glyphs that don't use backslash

ErikKarlen committed:
Adds BQN activity bar with glyphs

Adds a new activity bar on the side with glyphs that can be pressed to add them to the active editor.

ErikKarlen committed:
Pretty print code snippets

Update gen_snippets.js so it pretty prints the json output and rerun it to update snippets.code-snippets.

ErikKarlen committed:
Update glyphs_provider.ts indentation

Change indentation of glyphs_provider.ts to 2 spaces instead of 4 so it's the same as for extension.ts

ErikKarlen committed:
Update button sizes and colours

ErikKarlen committed:
Revert "Pretty print code snippets"

This reverts commit e4758a0e5bbe158913ac812a3cb28a97fa67c080.

ErikKarlen commented:
It seems running the gen_snippets script removes some snippet info so I reverted that commit.

ErikKarlen committed:
Update snippets.code-snippets

mk12 commented:
Thanks, this looks like a nice addition.

Is it possible to make the sidebar only show up when editing a BQN file? I suspect a lot of people who install this extension only use BQN every now and then (e.g. trying it out for Advent of Code one year), so it feels like a bit of an overreach for it to take up sidebar real estate all the time. If that's not possible, maybe it could be off by default with a command to toggle it, and some way of making it discoverable that isn't too obtrusive.

Also, can you run `npm format` and `npm lint` please?

And yes don't worry about gen_snippets.js script, it's out of date.

ErikKarlen committed:
Only show sidebar for .bqn files

ErikKarlen committed:
Adds keyboard shortcuts to tooltips

Shows the key combination needed to write each glyph in its button's tooltip.

Also fixed issues found with npm run lint and updated the formatting with npm run fmt.

ErikKarlen commented:
I added a "when" clause to package.json so it should only show up when editing .bqn files. I personally am planning to try doing Advent of Code using BQN to learn which is why I wanted to add this feature to make it easier for me to learn. Because of this I also added the keyboard shortcut needed to write each glyph to each button's tooltip.

<img width="224" height="118" alt="image" src="https://github.com/user-attachments/assets/1ceaa64d-2d66-4f72-82f2-593400080859" />

Now I also ran npm run lint and fixed the issues and npm run fmt to fix the formatting.

ErikKarlen committed:
Updates comment about prefix

mk12 commented:
Published on 0.2.0 on the VS Code Marketplace and Open VSX.
